reformer_pytorch/autopadder.py

```diff
@@ -12,21 +14,20 @@ def pad_to_multiple(tensor, seqlen, multiple, dim=-1):
     pad_offset = (0,) * (-1 - dim) * 2
     return F.pad(tensor, (*pad_offset, 0, remainder), value=0)
 
+
 class Autopadder(nn.Module):
-    def __init__(self, net):
+    def __init__(self, module):
         super().__init__()
-        assert isinstance(net, (Reformer, ReformerLM)), 'Only accepts Reformer or ReformerLM classes'
-        self.net = net
-
-        reformer = net.reformer if isinstance(net, ReformerLM) else net
-        self.pad_dim = -1 if isinstance(net, ReformerLM) else -2
 
-        self.bucket_size = reformer.bucket_size
-        self.num_mem_kv = reformer.num_mem_kv
-        self.full_attn_thres = reformer.full_attn_thres
+        allowed_classes = (Reformer, ReformerLM, LSHAttention, LSHSelfAttention)
+        assert isinstance(module, allowed_classes), \
+            'Autopadder only accepts Reformer, ReformerLM, ' \
+            'LSHAttention and LSHSelfAttention'
+        self.module = module
+        self.pad_dim = -1 if isinstance(module, ReformerLM) else -2
 
     def forward(self, x, **kwargs):
-        b, t, m = *x.shape[:2], self.num_mem_kv
+        b, t, m = *x.shape[:2], self.module.num_mem_kv
```

Comment:
it's tricky because depending on which module is set here, `num_mem_kv` may not be available

Reply:
perhaps it will just be cleaner to separate out Autopadder so that there is one individual one for each type of module